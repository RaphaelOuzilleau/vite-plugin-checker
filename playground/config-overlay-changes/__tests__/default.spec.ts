import { describe, expect, it } from 'vitest'
import {
  editFile,
  getHmrOverlay,
  getHmrOverlayText,
  isServe,
  pollingUntil,
  sleep,
  sleepForEdit,
  sleepForServerReady,
} from '../../testUtils'

describe('config-default', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial and following errors', async () => {
      await sleepForServerReady()
      const [message1, file1, frame1] = await getHmrOverlayText()
      expect(message1).toMatchSnapshot()
      expect(file1).toMatchSnapshot()
      expect(frame1).toMatchSnapshot()

      console.log('-- overlay update after edit --')
      editFile('src/main.ts', (code) => code.replace('Hello', 'Hello1'))
      await sleepForEdit()
      await pollingUntil(getHmrOverlay, (dom) => !!dom)
      const [, , frame2] = await getHmrOverlayText()
      expect(frame2).toMatchSnapshot()

      console.log('-- overlay dismiss after fix error --')
      editFile('src/main.ts', (code) => code.replace('var hello', `const hello`))
      editFile('src/main.ts', (code) => code.replace('! as', ` as`))
      await sleep(6000)
      await expect(getHmrOverlayText()).rejects.toThrow(
        'Invariant failed: <vite-plugin-checker-error-overlay> shadow dom is expected to be found, but got null'
      )
    })
  })
})
