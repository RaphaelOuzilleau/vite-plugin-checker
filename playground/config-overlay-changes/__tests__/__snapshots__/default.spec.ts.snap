// Vitest Snapshot v1

exports[`config-default > serve > get initial and following errors 1`] = `"Unexpected var, use let or const instead. (no-var)"`;

exports[`config-default > serve > get initial and following errors 2`] = `"<PROJECT_ROOT>/playground-temp/config-overlay-changes/src/main.ts:3:1"`;

exports[`config-default > serve > get initial and following errors 3`] = `
"    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello'
      | ^^^^^^^^^^^^^^^^^^^
    4 |
    5 | const rootDom = document.querySelector('#root')! as HTMLElement
    6 | rootDom.innerHTML = hello + text"
`;

exports[`config-default > serve > get initial and following errors 4`] = `
"    1 | import { text } from './text'
    2 |
  > 3 | var hello = 'Hello1'
      | ^^^^^^^^^^^^^^^^^^^^
    4 |
    5 | const rootDom = document.querySelector('#root')! as HTMLElement
    6 | rootDom.innerHTML = hello + text"
`;
